File: src/main/java/com/openpms/server/user/dto/CreateBody.java
package com.openpms.server.user.dto;

import com.alibaba.fastjson2.annotation.JSONField;
import com.alibaba.fastjson2.annotation.JSONType;
import lombok.Data;

@Data
@JSONType(schema = """
        {
            "required": ["email", "password"]
        }""")
public class CreateBody {
    @JSONField(schema = "{'format':'email'}")
    private String email;
    @JSONField(schema = "{'minLength': '8'}")
    private String password;
    private String name;
    private String avatar;
    private Boolean status = true;
}

File: src/main/kotlin/com/creator/server/user/dto/CreateBody.kt
package com.creator.server.user.dto

import com.alibaba.fastjson2.annotation.JSONField
import com.alibaba.fastjson2.annotation.JSONType

@JSONType(
    schema = """
    {
        "required": ["email", "password"]
    }
    """
)
data class CreateBody(
    @JSONField(schema = "{'format':'email'}")
    val email: String,
    @JSONField(schema = "{'minLength': '8'}")
    val password: String,
    val name: String? = null,
    val avatar: String? = null,
    val status: Boolean = true
)
